jwriter/token_writer_default.go

```diff
@@ -0,0 +1,191 @@
+// +build !launchdarkly_easyjson
+
+package jwriter
+
+import (
+	"encoding/json"
+	"io"
+	"strconv"
+	"unicode/utf8"
+)
+
+// This file defines the default implementation of the low-level JSON token writer. If the launchdarkly_easyjson
+// build tag is enabled, we use the easyjson adapter in token_writer_easyjson.go instead. These have the same
+// methods so the Writer code does not need to know which implementation we're using; however, we don't
+// actually define an interface for these, because calling the methods through an interface would limit
+// performance.
+
+var (
+	tokenNull  = []byte("null")  //nolint:gochecknoglobals
+	tokenTrue  = []byte("true")  //nolint:gochecknoglobals
+	tokenFalse = []byte("false") //nolint:gochecknoglobals
+)
+
+type tokenWriter struct {
+	buf       streamableBuffer
+	tempBytes [50]byte
```

Comment:
Is this magic number derived from some particular formula or is it just a reasonable guess of the needed size?

Reply:
If you look at where it's actually used, this is for converting ints and floats to strings, and also for building hex escape sequences like `\u0000`. The only one of those where the maximum length is indeterminate is floats, so I left some excess space there, although if `strconv.AppendFloat` runs out of room it'll just allocate a new buffer.